### ISSUE TITLE
Rotating of lines not possible

### ISSUE THREAD

**Issue by tordans** [bryceco/GoMap#456]
It would be great, if I could rotate lines the same way as areas.

Location: https://www.openstreetmap.org/way/624191234

Step by Step: I think I copied the element from somewhere nearby and then tried to rotate it in position.

Screenshot: No "rotate" Option
![image](https://user-images.githubusercontent.com/111561/94735854-a7691d80-036b-11eb-8d12-67115df71989.png)

**Comment by bryceco** [comment]
It's trivial to support this, the only question is whether its an operation that is common enough to add a menu entry. I've honestly never had reason to do this.

**Comment by tordans** [comment]
Agreed, it's mainly for the specific case when the line is copied and not newly drawn, so its not where its supposed to be `+` "complex" (has many nodes), so it cannot be easy moved node by node.

On other case where I might use this is, to work around the auto-snapping. I regularly have the issue, that my initial new line is just slightly out of place but once I move the node, it snaps which I don't want in those cases. Rotating would be a way out. Example: This railing which would snap to the landuse https://www.openstreetmap.org/way/851516440.

**Comment by bryceco** [comment]
The snapping is definitely a double-edged sword, usually helpful but sometimes annoying. Are there instances where it snaps without a visual indication it will snap? 

I'm not sure how when you would use rotation to avoid snapping. 

I feel like the current way menu is short enough that I don't mind adding rotate but only when the entire object is visible: I wouldn't support it for a highway that extends off-screen.

**Comment by tordans** [comment]
I was looking for this feature again, recently. I mapped a bollard_row which had to be added on every side of the intersection. It is hard to get good near-90° angles for the line, so that is some "fidgeting". Afterwards I wanted to duplicate and rotate the geometry…

<img width="391" alt="image" src="https://github.com/bryceco/GoMap/assets/111561/ea2c4148-2080-4b07-ab6f-6d6e40e73d3c">
